### PR TITLE
[prometheus-statsd-exporter] add podLabels to add extra labels to pods

### PR DESCRIPTION
**Pull request by clavinjune**
#### What this PR does / why we need it

This PR enables additional labels for statsd-exporter pods

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

#### Mentioning
- @scDisorder

### CODE REVIEW

**Comment by clavinjune** [comment]
#### values.yaml

```diff
 # Extra labels to be added to pods
-podLabels: {}
+podLabels:
+  exampleLabels: exampleValue
```

#### result

```yaml
---
# Source: prometheus-statsd-exporter/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.7.1
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
    exampleLabels: exampleValue
---
# Source: prometheus-statsd-exporter/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.7.1
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
    exampleLabels: exampleValue
spec:
  type: ClusterIP
  ports:
    - port: 9102
      targetPort: http
      protocol: TCP
      name: http
    - port: 9125
      targetPort: statsd-tcp
      protocol: TCP
      name: statsd-tcp
    - port: 9125
      targetPort: statsd-udp
      protocol: UDP
      name: statsd-udp
  selector:
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
---
# Source: prometheus-statsd-exporter/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.7.1
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
    exampleLabels: exampleValue
spec:
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/name: prometheus-statsd-exporter
      app.kubernetes.io/instance: release-name
  template:
    metadata:
      labels:
        helm.sh/chart: prometheus-statsd-exporter-0.7.1
        app.kubernetes.io/name: prometheus-statsd-exporter
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "v0.22.8"
        app.kubernetes.io/managed-by: Helm
        exampleLabels: exampleValue
    spec:
      serviceAccountName: release-name-prometheus-statsd-exporter
      securityContext:
        {}
      containers:
        - name: prometheus-statsd-exporter
          securityContext:
            {}
          image: "prom/statsd-exporter:v0.22.8"
          imagePullPolicy: IfNotPresent
          args:
            - --web.listen-address=:9102
            - --web.telemetry-path=/metrics
            - --statsd.listen-udp=:9125
            - --statsd.listen-tcp=:9125
            - --statsd.cache-size=1000
            - --statsd.event-queue-size=10000
            - --statsd.event-flush-threshold=1000
            - --statsd.event-flush-interval=200ms
          ports:
            - name: http
              containerPort: 9102
              protocol: TCP
            - name: statsd-tcp
              containerPort: 9125
              protocol: TCP
            - name: statsd-udp
              containerPort: 9125
              protocol: UDP
          livenessProbe:
            httpGet:
              path: /health
              port: http
          readinessProbe:
            httpGet:
              path: /health
              port: http
          resources:
            {}
---
# Source: prometheus-statsd-exporter/templates/tests/test-connection.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "release-name-prometheus-statsd-exporter-test-connection"
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.7.1
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
    exampleLabels: exampleValue
  annotations:
    "helm.sh/hook": test
spec:
  containers:
    - name: wget
      image: busybox
      command: ['wget']
      args: ['release-name-prometheus-statsd-exporter.observability.svc.cluster.local:9102/metrics']
  restartPolicy: Never
```

**Comment by clavinjune** [comment]
bumping this up 🙇

**Review by scDisorder — DISMISSED** [review]

**Comment by clavinjune** [comment]
I think this PR needs to be approved by another maintainer before it can be merged 🤔 

mentioning approver from the latest merged PR:
- @monotek

File: charts/prometheus-statsd-exporter/Chart.yaml
Comment (monotek) [inline charts/prometheus-statsd-exporter/Chart.yaml:4]: new feature should change minor version

File: charts/prometheus-statsd-exporter/templates/deployment.yaml
Comment (monotek) [inline charts/prometheus-statsd-exporter/templates/deployment.yaml:22]: please keep it as it is

File: charts/prometheus-statsd-exporter/templates/_helpers.tpl
Comment (monotek) [inline charts/prometheus-statsd-exporter/templates/_helpers.tpl:43]: please move this to selectorlabels

**Comment by clavinjune** [comment]
**Values.yaml**

```diff
 # Extra labels to be added to pods
-podLabels: {}
+podLabels:
+  exampleLabels: exampleValue
```

**result**

```yaml
---
# Source: prometheus-statsd-exporter/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.8.0
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    exampleLabels: exampleValue
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
---
# Source: prometheus-statsd-exporter/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.8.0
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    exampleLabels: exampleValue
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
    - port: 9102
      targetPort: http
      protocol: TCP
      name: http
    - port: 9125
      targetPort: statsd-tcp
      protocol: TCP
      name: statsd-tcp
    - port: 9125
      targetPort: statsd-udp
      protocol: UDP
      name: statsd-udp
  selector:
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    exampleLabels: exampleValue
---
# Source: prometheus-statsd-exporter/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-prometheus-statsd-exporter
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.8.0
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    exampleLabels: exampleValue
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/name: prometheus-statsd-exporter
      app.kubernetes.io/instance: release-name
      exampleLabels: exampleValue
  template:
    metadata:
      labels:
        app.kubernetes.io/name: prometheus-statsd-exporter
        app.kubernetes.io/instance: release-name
        exampleLabels: exampleValue
    spec:
      serviceAccountName: release-name-prometheus-statsd-exporter
      securityContext:
        {}
      containers:
        - name: prometheus-statsd-exporter
          securityContext:
            {}
          image: "prom/statsd-exporter:v0.22.8"
          imagePullPolicy: IfNotPresent
          args:
            - --web.listen-address=:9102
            - --web.telemetry-path=/metrics
            - --statsd.listen-udp=:9125
            - --statsd.listen-tcp=:9125
            - --statsd.cache-size=1000
            - --statsd.event-queue-size=10000
            - --statsd.event-flush-threshold=1000
            - --statsd.event-flush-interval=200ms
          ports:
            - name: http
              containerPort: 9102
              protocol: TCP
            - name: statsd-tcp
              containerPort: 9125
              protocol: TCP
            - name: statsd-udp
              containerPort: 9125
              protocol: UDP
          livenessProbe:
            httpGet:
              path: /health
              port: http
          readinessProbe:
            httpGet:
              path: /health
              port: http
          resources:
            {}
---
# Source: prometheus-statsd-exporter/templates/tests/test-connection.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "release-name-prometheus-statsd-exporter-test-connection"
  labels:
    helm.sh/chart: prometheus-statsd-exporter-0.8.0
    app.kubernetes.io/name: prometheus-statsd-exporter
    app.kubernetes.io/instance: release-name
    exampleLabels: exampleValue
    app.kubernetes.io/version: "v0.22.8"
    app.kubernetes.io/managed-by: Helm
  annotations:
    "helm.sh/hook": test
spec:
  containers:
    - name: wget
      image: busybox
      command: ['wget']
      args: ['release-name-prometheus-statsd-exporter.observability.svc.cluster.local:9102/metrics']
  restartPolicy: Never
```

**Pull request merged**
